### PR TITLE
feat: trtllm conditional disaggregation

### PR DESCRIPTION
**Pull request by shpgy-shpgy**
#### Overview:
Add support for Conditional Disaggregation to TensorRT-LLM.
Route to PD worker or non-PD worker depending on input request length.

#### Where should the reviewer start?
components/src/dynamo/trtllm/request_handlers/handlers.py
lib/llm/src/kv_router/scheduler.rs

## Summary by CodeRabbit

* New Features
  * Conditional disaggregation with a configurable short-prefill threshold so small requests can be handled locally to reduce latency.
  * Engine runtime can store and use a selectable disaggregation mode.
  * Optional ISL-based routing threshold to refine which workers contribute logits.

* Bug Fixes
  * Decode no longer errors when disaggregated parameters are missing; it logs and falls back to local prefill.

### CODE REVIEW

**Comment by PeaBrane** [comment]
@shpgy-shpgy 
Thanks for your contribution. It is unlikely we plan to support cond disagg at the Python engine handler level. The current roadmap is for us bake the cond disagg logic directly into the Rust frontend, more specifically in prefill_router.rs, which is still WIP.

For future reference, it would be better for us to first open a feature issue to discuss with the team, to make sure everyone is aligned.

**Comment by dagil-nvidia** [comment]
Closing due to misalignment with Dynamo conditional disagg approach
